fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PolymorphicFunctionAnalyzer.java

```diff
@@ -144,11 +146,19 @@ public Type apply(Type[] types) {
         }
     }
 
-    private static final ImmutableMap<String, java.util.function.Function<Type[], Type>> DEDUCE_RETURN_TYPE_FUNCTIONS
-            = ImmutableMap.<String, java.util.function.Function<Type[], Type>>builder()
-            .put("map_keys", new MapKeysDeduce())
-            .put("map_values", new MapValuesDeduce())
-            .build();
+    private static class NamedStructDeduce implements java.util.function.Function<Type[], Type> {
+        @Override
+        public Type apply(Type[] types) {
+            return new StructType(IntStream.range(0, types.length / 2).boxed().map(i -> types[i * 2 + 1])
```

Comment:
Because it is a named_struct, I think you need to construct a NAMED struct for this function return type rather than an unnamed struct.